net/ip_utils.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+const IPV4_REGEX = /^[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}$/;
+
+export type AddressType = "IPv4" | "IPv6";
+
+function ipToInt(addr: string): number {
+  return addr.split(".").reduce(
+    (acc, octet) => (acc << 8) + parseInt(octet),
+    0,
+  ) >>> 0;
+}
+
+/**
+ * Returns the address type ("IPv4" or "IPv6") for a given remote address string.
+ *
+ * @param remoteAddr - The remote address as a string.
+ * @returns The address type ("IPv4" | "IPv6") or undefined if not recognized.
+ */
+export const distinctRemoteAddr = (
+  remoteAddr: string,
+): AddressType | undefined => {
+  if (isIPv4(remoteAddr)) {
+    return "IPv4";
+  }
+  if (isIPv6(remoteAddr)) {
+    return "IPv6";
+  }
+};
```

Comment:
IMHO I have a feeling this function simply isn't needed and adding it now might be premature. If it were up to me, I'd wait till there's sufficient demand for such a function before adding it to the API.